src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    Animal.alive.push(this);
+    this.health = 100;
+    this.name = name;
+  }
 }
 
+Animal.alive = [];
+
+Animal.reduceAllDeadAnimals = () => {
+  Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+};
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    const isHerbivore = animal instanceof Herbivore;
+    const isAlive = Animal.alive.includes(animal);
+    const isNotHidding = !animal.hidden;
```

Comment:
better to start with positive in booleans, it is easier to read for the human mind
```suggestion
    const isHidding = animal.hidden;
```